projects/extension/src/background/ConnectionManager.ts

```diff
@@ -244,195 +163,152 @@ export class ConnectionManager
    *
    * @returns addedChain - An the newly added chain info
    */
-  async addChain(
+  addChain(
     chainSpec: string,
     jsonRpcCallback?: JsonRpcCallback,
     tabId?: number,
-  ): Promise<Network> {
+  ): Promise<Chain> {
```

Comment:
Not a problem, but I'm surprised that this works, I thought you had to always use both `async` and `Promise`.

Reply:
Nope, you can return a `Promise` without using `async`. There are many instances when that's actually necessary/preferable. On the other hand, creating an `async` function without using `await` is "dangerous" because you may inadvertently creating a `Promise` for code that otherwise could be synchronous.